packages/common/src/contract/kuai.loader.ts

```diff
@@ -1,11 +1,21 @@
 import type { ContractDeploymentInfo, ContractLoader } from './interface'
 import fs from 'node:fs'
+import { validateName, validatePath, validateScriptBase, validatecellDep } from './validator'
 
 export class KuaiContractLoader implements ContractLoader {
   constructor(private path: string) {}
 
   load = (): ContractDeploymentInfo[] =>
     fs.existsSync(this.path) ? JSON.parse(fs.readFileSync(this.path, 'utf-8')) ?? [] : []
 
-  write = (info: ContractDeploymentInfo[]) => fs.writeFileSync(this.path, Buffer.from(JSON.stringify(info)))
+  write = (info: ContractDeploymentInfo[]) => {
+    if (!Array.isArray(info)) {
+      throw new Error(`info must be an array`)
+    }
+    for (const item of info) {
+      validatecellDep(validatePath(validateName(validateScriptBase(validatecellDep(item)))))
```

Comment:
This usage is a bit weird. Each checker returns `item` directly, which is called monad, so it can be refactored into

```js
item
|> validatecellDep
|> validatePath
|> validateName
```

But the pipeline operator in JavaScript has not been approved, so it is usually in OOP form

```js
item.validatecellDep().validatePath().validateName()
```

However, in this case the `item` will not be updated in each checker. It can be simply executed separately

```js
validatecellDep(item)
validatePath(item)
validateName(item)
```

---

BTW, `validatecellDep` is called twice in this list. Maybe one of them should be `validateCellDeps`